src/framework/mpas_io.F

```diff
@@ -1984,22 +1993,40 @@ subroutine MPAS_io_get_var_generic(handle, fieldname, intVal, intArray1d, intArr
 !         call mpas_log_write('  value is char')
 
          charVal_p => charVal
+
+         ! get the dimension of the char variable to ensure the provided output buffer is large enough
+         call MPAS_io_inq_var(handle, fieldname, fieldtype, ndims, dimnames, dimsizes, local_ierr)
+         do i = 1, ndims
+            message = '  MPAS_io_get_var_generic len(charVal):$i var "'//trim(fieldname)// &
+                      '" type is $i dim $i is '// trim(dimnames(i))//' size is $i'
+            IO_DEBUG_WRITE(message , intArgs=(/len(charVal), fieldtype, i, dimsizes(i)/))
+         end do
+         ! because charVal is provided, assume dimension 1 is the string length
+         if (dimsizes(1) > len(charVal)) then
+            local_ierr = MPAS_IO_ERR_INSUFFICIENT_ARG
+            message = '  MPAS_io_get_var_generic var "'//trim(fieldname)//  &
```

Comment:
Are the leading two spaces in this string intentional? They lead to messages that look like this:
```
ERROR:   MPAS_io_get_var_generic var "xtime" len too big, len=128 buflen=64
ERROR: MPAS IO Error: Attempting to read a string into a buffer which is too small.
```